### PR TITLE
Fixing filter cache issues

### PR DESCRIPTION
**Pull request by tavinathanson**
Could probably use more tests, but how does this look for now?

### CODE REVIEW

**Comment by tavinathanson** [comment]
@jburos added another fix: checking for `None` to avoid calling `_hash_filter_fn` on `None`.

File: cohorts/cohort.py [hunk 1]
```diff
@@ -554,12 +554,15 @@ def _load_single_patient_variants(self, patient, filter_fn, use_cache=True, **kw
...
         """
-        filter_fn_name = self._get_function_name(filter_fn)
-        logger.debug("loading variants for patient {} with filter_fn {}".format(patient.id, filter_fn_name))
-        use_filtered_cache = use_cache
-        if sys.version_info < (3, 3):
-            logger.info("... disabling filtered cache due to python version")
+        if filter_fn is None:
             use_filtered_cache = False
+        else:
+            filter_fn_name = self._get_function_name(filter_fn)
```
Comment (tavinathanson) [inline cohorts/cohort.py:560]: No changes in this block, just moved

**Comment by tavinathanson** [comment]
Can't wait for https://github.com/hammerlab/cohorts/issues/175!

**Comment by tavinathanson** [comment]
Fixes https://github.com/hammerlab/cohorts/issues/195

**Comment by tavinathanson** [comment]
@jburos oops, looks like this one was forgotten by both of us; can you take a look?

**Review by jburos — APPROVED** [review]
LGTM - thanks for fixing my bugs :)

**Pull request merged**